lib/factory.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+class Factory
+  def self.new(*arguments, &block)
+    if arguments.first.is_a?(String)
+      raise NameError unless arguments.first.match?(/^[A-Z]/)
+      class_name = arguments.first
+      members_names = arguments[1..-1]
+    else
+      members_names = arguments
+    end
+
+    class_implementetion = Class.new do
+      attr_accessor *members_names
+
+      class_eval(&block) if block_given?
+
+      define_method :members do
+        members_names.each(&:to_sym)
+      end
+
+      def initialize(*members_values)
+        raise ArgumentError if members_values.length > members.length
+        members.zip(members_values).each do |name, value|
+          instance_variable_set("@#{name}", value)
```

Comment:
Can we use `send` for better performance?

Reply:
We can use send for better performance. I'll fix it